src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -133,32 +146,102 @@ SecurityAnalysisReport runSync(SecurityAnalysisParameters securityAnalysisParame
+            List<LfNetworkList> lfNetworksList = new ArrayList<>();
+
+            Lock networkLock = new ReentrantLock();
+            List<CompletableFuture<Void>> futures = new ArrayList<>();
+            for (int i = 0; i < contingenciesPartitions.size(); i++) {
+                final int partitionNum = i;
+                var contingenciesPartition = contingenciesPartitions.get(i);
+                futures.add(CompletableFuture.runAsync(() -> {
+
+                    var partitionTopoConfig = new LfTopoConfig(topoConfig);
+
+                    //  we have to pay attention with IIDM network multi threading even when allowVariantMultiThreadAccess is set:
+                    //    - variant cloning and removal is not thread safe
+                    //    - we cannot read or write on an exising variant while another thread clone or remove a variant
+                    //    - be aware that even after LF network loading, though LF network we get access to original IIDM
+                    //      variant (for instance to get reactive capability curve), so allowVariantMultiThreadAccess mode
+                    //      is absolutely required
+                    //  so in order to be thread safe, we need to:
+                    //    - lock LF network creation (which create a working variant, see {@code LfNetworkList}
+                    //    - delay {@code LfNetworkList} closing (which remove a working variant) out of worker thread
+                    LfNetworkList lfNetworks;
+                    List<PropagatedContingency> propagatedContingencies;
+                    P parameters;
+                    networkLock.lock();
+                    try {
+                        network.getVariantManager().setWorkingVariant(workingVariantId);
+
+                        propagatedContingencies = PropagatedContingency.createList(network, contingenciesPartition, partitionTopoConfig, creationParameters);
+
+                        parameters = createParameters(lfParameters, lfParametersExt, partitionTopoConfig.isBreaker());
+
+                        // create networks including all necessary switches
+                        lfNetworks = Networks.load(network, parameters.getNetworkParameters(), partitionTopoConfig, saReportNode);
+                        lfNetworksList.add(0, lfNetworks); // FIXME to workaround variant removal bug, to fix in core
+                    } finally {
+                        networkLock.unlock();
+                    }
 
-            return new SecurityAnalysisReport(result);
+                    // run simulation on largest network
+                    partitionResults.set(partitionNum, lfNetworks.getLargest().filter(n -> n.getValidity() == LfNetwork.Validity.VALID)
+                            .map(largestNetwork -> runSimulations(largestNetwork, propagatedContingencies, parameters, securityAnalysisParameters, operatorStrategies, actions, limitReductions))
+                            .orElse(createNoResult()));
```

Comment:
Isn't it easier here to use a Callable that using the synchronized list to store the resutls ?